src/sanitation.js

```diff
@@ -247,18 +280,22 @@ export function isValidAttr(
   }
 
   // Remove blacklisted attributes from specific tags e.g. input[formaction].
-  const attrBlacklist = isAmp4Email(doc)
-    ? EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName]
-    : BLACKLISTED_TAG_SPECIFIC_ATTRS[tagName];
+  const attrBlacklist = Object.assign(
+    map(),
+    BLACKLISTED_TAG_SPECIFIC_ATTRS,
+    isAmp4Email(doc) ? EMAIL_BLACKLISTED_TAG_SPECIFIC_ATTRS : {}
```

Comment:
nit can we save a result of the isAmp4Email(doc) call to be used here and below.